src/libstd/sys/windows/c.rs

```diff
@@ -1023,6 +1028,56 @@ pub struct timeval {
     pub tv_usec: c_long,
 }
 
+// Functions forbidden when targeting UWP
+#[cfg(not(target_os = "uwp"))]
+ifdef! {
+extern "system" {
+    pub fn ReadConsoleW(hConsoleInput: HANDLE,
+                        lpBuffer: LPVOID,
+                        nNumberOfCharsToRead: DWORD,
+                        lpNumberOfCharsRead: LPDWORD,
+                        pInputControl: PCONSOLE_READCONSOLE_CONTROL) -> BOOL;
+
+    pub fn WriteConsoleW(hConsoleOutput: HANDLE,
+                         lpBuffer: LPCVOID,
+                         nNumberOfCharsToWrite: DWORD,
+                         lpNumberOfCharsWritten: LPDWORD,
+                         lpReserved: LPVOID) -> BOOL;
+
+    pub fn GetConsoleMode(hConsoleHandle: HANDLE,
+                          lpMode: LPDWORD) -> BOOL;
+    pub fn GetFileInformationByHandle(hFile: HANDLE,
+                        lpFileInformation: LPBY_HANDLE_FILE_INFORMATION)
+                        -> BOOL;
+    pub fn SetHandleInformation(hObject: HANDLE,
+                                dwMask: DWORD,
+                                dwFlags: DWORD) -> BOOL;
+    pub fn AddVectoredExceptionHandler(FirstHandler: ULONG,
+                                       VectoredHandler: PVECTORED_EXCEPTION_HANDLER)
+                                       -> LPVOID;
+    pub fn CreateHardLinkW(lpSymlinkFileName: LPCWSTR,
+                           lpTargetFileName: LPCWSTR,
+                           lpSecurityAttributes: LPSECURITY_ATTRIBUTES)
+                           -> BOOL;
+    #[link_name = "SystemFunction036"]
+    pub fn RtlGenRandom(RandomBuffer: *mut u8, RandomBufferLength: ULONG) -> BOOLEAN;
+}
+}
+
+// UWP specific functions
+#[cfg(target_os = "uwp")]
+ifdef! {
+extern "system" {
+    pub fn GetFileInformationByHandleEx(hFile: HANDLE,
+                                        fileInfoClass: FILE_INFO_BY_HANDLE_CLASS,
+                                        lpFileInformation: LPVOID,
+                                        dwBufferSize: DWORD) -> BOOL;
+    pub fn BCryptGenRandom(hAlgorithm: LPVOID, pBuffer: *mut u8,
+                           cbBuffer: ULONG, dwFlags: ULONG) -> LONG;
```

Comment:
Out of curiosity are these available on normal Windows OSes? If so it may be reasonable to just leave them here with an `allow(dead_code)` annotation.

Reply:
They can always be left in, because unused extern functions never have any code generated for them and so linking is never attempted for them.

Reply:
They are both available in Vista or newer systems for msvc and mingw-w64 BTW.